nucleus/payara-modules/nucleus-microprofile/config-service/src/main/java/fish/payara/nucleus/microprofile/config/spi/PayaraConfig.java

```diff
@@ -177,74 +174,60 @@ private String getValue(String propertyName) {
+        // search for a matching raw type
+        for (Entry<Type, Converter> entry : converters.entrySet()) {
+            Type type1 = entry.getKey();
+            if (type1 instanceof ParameterizedType) {
+                ParameterizedType ptype = (ParameterizedType)type1;
+                if (ptype.getRawType().equals(propertyType)) {
+                    return entry.getValue();
                 }
             }
         }
         
-        if (converter == null) {
-            // see if a common sense converter can be created
-            boolean commonSense = true;
-            try {
-                Method m = propertyType.getMethod("of", String.class);
-                converter = (Converter<T>) new CommonSenseConverter(m);
-                converters.put(propertyType, converter);
-                return converter;
-            } catch (NoSuchMethodException | SecurityException ex) {
-                try {
-                    Method m = propertyType.getMethod("valueOf",String.class);
-                    converter = (Converter<T>) new CommonSenseConverter(m);
-                    converters.put(propertyType, converter);
-                    return converter;
-                } catch (NoSuchMethodException | SecurityException ex1) {
-                    try {
-                        Constructor c = propertyType.getConstructor(String.class);
-                        converter = (Converter<T>) new CommonSenseConverter(c);
-                        converters.put(propertyType, converter);
-                        return converter;
-                    } catch (NoSuchMethodException | SecurityException ex2) {
-                        try {
-                            Method m = propertyType.getMethod("parse", CharSequence.class);
-                            converter = (Converter<T>) new CommonSenseConverter(m);
-                            converters.put(propertyType, converter);
-                            return converter;
-                        } catch (NoSuchMethodException | SecurityException ex3) {
-                            commonSense = false;
-                        }
-                    }
-                }
-            }
+        // see if a common sense converter can be created
+        Optional<Converter<T>> commonSenseConverter = CommonSenseConverter.forClass(propertyType);
+        if (commonSenseConverter.isPresent()) {
+            converters.put(propertyType, commonSenseConverter.get());
+            return commonSenseConverter.get();
         }
-        if (converter == null) {
-            throw new IllegalArgumentException("Unable to convert value to type " + propertyType.getCanonicalName());
+
+        throw new IllegalArgumentException("Unable to convert value to type " + propertyType.getCanonicalName());
+    }
+
+
+
+    private static <T> Class<T> boxedTypeOf(Class<T> type) {
+        if (!type.isPrimitive()) {
+            return type;
+        } else if (type.equals(int.class)) {
```

Comment:
`else` can be spared - I know it makes the method longer with our formatting rules but it is simpler without the `else`. `equals` can be `==`.

I benchmarked this years ago - using a map is surprisingly equally fast but I also ended up keeping the if's since those can be ordered checking for the most likely first.

Reply:
Since this ends up just being key in the map, I think I'll just initialize default converters map with both types.